src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNames.cs

```diff
@@ -0,0 +1,261 @@
+                    {
+                        ReportPropertyDiagnostic(context, PropertyNameRule, ParameterDiagnosticReason.NameMismatch, param, prop);
+                    }
+
+                    if (!prop.IsPublic())
+                    {
+                        ReportPropertyDiagnostic(context, PropertyPublicRule, ParameterDiagnosticReason.PropertyInappropriateVisibility, param, prop);
+                    }
+                }
+            }
+
+            private static bool IsSupportedProp([NotNullWhen(true)] IPropertySymbol? prop)
+            {
+                if (prop == null)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            private static bool IsSupportedField([NotNullWhen(true)] IFieldSymbol? field)
+            {
+                if (field == null)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            private static bool IsParamMatchFieldName(IParameterSymbol param, IFieldSymbol field)
+            {
+                if (param.Name.Length != field.Name.Length)
+                {
+                    return false;
+                }
+
+                var paramWords = WordParser.Parse(param.Name, WordParserOptions.SplitCompoundWords);
+                var fieldWords = WordParser.Parse(field.Name, WordParserOptions.SplitCompoundWords).ToImmutableArray();
+
+                return paramWords.All(x => WordParser.ContainsWord(x, WordParserOptions.SplitCompoundWords, fieldWords));
+            }
+
+            private static bool IsParamMatchPropName(IParameterSymbol param, IPropertySymbol prop)
+            {
+                if (param.Name.Length != prop.Name.Length)
+                {
+                    return false;
+                }
+
+                var paramWords = WordParser.Parse(param.Name, WordParserOptions.SplitCompoundWords);
+                var propWords = WordParser.Parse(prop.Name, WordParserOptions.SplitCompoundWords).ToImmutableArray();
+
+                return paramWords.All(x => WordParser.ContainsWord(x, WordParserOptions.SplitCompoundWords, propWords));
+            }
+
+            private bool IsJsonConstructor([NotNullWhen(returnValue: true)] IMethodSymbol? method)
+            => method.IsConstructor() &&
+                method.HasAttribute(this._jsonConstructorAttributeInfoType);
+
+            public bool ShouldAnalyzeMethod(IMethodSymbol method)
+            {
+                // We only care about constructors with parameters.
+                if (method.Parameters.IsEmpty)
+                {
+                    return false;
+                }
+
+                // We only care about constructors that are marked with JsonConstructor attribute.
+                if (!this.IsJsonConstructor(method))
+                {
+                    return false;
+                }
+
+                return true;
```

Comment:
nit:

```suggestion
                return IsJsonConstructor(method);
```

Reply:
I prefer doing such small refactorings at the latest stages of development. Though, it makes sense, thanks.

Reply:
Done.